org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLRename.java

```diff
@@ -0,0 +1,279 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4xml.services;
+
+import static org.eclipse.lsp4xml.services.XMLHighlighting.covers;
+import static org.eclipse.lsp4xml.services.XMLHighlighting.doesTagCoverPosition;
+import static org.eclipse.lsp4xml.services.XMLHighlighting.getTagNameRange;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.annotation.Nullable;
+
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4xml.commons.BadLocationException;
+import org.eclipse.lsp4xml.dom.DOMAttr;
+import org.eclipse.lsp4xml.dom.DOMDocument;
+import org.eclipse.lsp4xml.dom.DOMElement;
+import org.eclipse.lsp4xml.dom.DOMNode;
+import org.eclipse.lsp4xml.dom.parser.TokenType;
+
+/**
+ * XMLRename
```

Comment:
Explain more what is the goal of this class like rename namespaces and add @author

Reply:
Before for the Rename feature, it was using XMLHighlighting to get Highlights and then it turned them into TextEdits. Also Highlighting would only work for 1 Element (Start Tag and End Tag) but with XMLRename it will handle many elements with the same name, namespaces and maybe other renaming in the future.

Reply:
Ok thanks for the clarification